core/azure-core/src/main/java/com/azure/core/exception/ServiceRequestException.java

```diff
@@ -3,14 +3,20 @@
 
 package com.azure.core.exception;
 
+import com.azure.core.http.HttpRequest;
+
 /**
- * An exception thrown for an invalid response with custom error information.
+ * The exception occurred while attempting to connect a socket to a azure service address and port.
+ * Typically, the connection was refused remotely (e.g., no process is listening on the azure service address/port).
  *
- * @see AzureException
+ * These errors are safe to retry.
  */
 public class ServiceRequestException extends AzureException {
 
-
+    /**
+     * Information about the associated HTTP response.
+     */
+    private HttpRequest request;
```

Comment:
The ServiceResponseException class was changed to not contain the HttpResponse information. It now feels a little odd that the ServiceRequestException includes the HttpRequest information. What do you think?

Reply:
I am ok to remove this out. I added this for checking what the failed request is. I can remove for now, since it is not that hard to add it back if needed.

Reply:
My inclination, rather than remove the HttpRequest field, would have been to rename the ServiceRequestException class to instead be HttpRequestException.